search/tasks.py

```diff
@@ -35,3 +74,127 @@ def update_document_with_partial(doc_id, partial_data):
 def increment_document_integer_field(doc_id, field_name, incr_amount):
     """Task that makes a request to increment some integer field in an ES document"""
     api.increment_document_integer_field(doc_id, field_name, incr_amount)
+
+
+@app.task(autoretry_for=(RetryException, ), retry_backoff=True, rate_limit='600/m')
+def index_post(api_username, post_id):
+    """
+    Index a post and its comments
+
+    Args:
+        api_username (str): The API username
+        post_id (str): The post string
+    """
+    with wrap_retry_exception(PrawcoreException, PRAWException):
+        api.index_post(api_username, post_id)
+
+
+@app.task(bind=True, autoretry_for=(RetryException, ), retry_backoff=True, rate_limit='600/m')
+def index_channel(self, api_username, channel_name):
+    """
+    Index the channel
+
+    Args:
+        api_username (str): API username
+        channel_name (str): The name of the channel
+    """
+    with wrap_retry_exception(PrawcoreException, PRAWException):
+        from channels.api import Api
+
+        client = Api(User.objects.get(username=api_username))
+        channel = client.get_channel(channel_name)
+        post_ids = [post.id for post in channel.new()]
```

Comment:
Seems like there should be an added channel API method for this. `get_channel_posts` or something like that

Reply:
There is actually, I used `list_posts`